libretro-common/rthreads/single_threaded.c

```diff
@@ -0,0 +1,29 @@
+#include <rthreads/rthreads.h>
+
+uintptr_t sthread_get_thread_id(sthread_t *thread)
+{
+   if (!thread)
+      return 0;
+   return (uintptr_t)1;
+}
+
+void slock_unlock(slock_t *lock)
+{
+}
+
+void slock_lock(slock_t *lock)
+{
+}
+
+void slock_free(slock_t *lock)
+{
+   if (!lock)
+      return;
+
+   free(lock);
+}
+
+slock_t *slock_new(void)
+{
+   return (slock_t*)calloc(1, 1);
```

Comment:
There is no need to waste memory with this, since nothing is done with the returned data.

`return (slock_t*)-1;`

Reply:
I disagree. The client code may check that 2 locks are the same by comparing pointers, so it's better not to alias 2 mutexes.

Reply:
What's the use case for that? In all my years of programming I've never done that.

https://github.com/libretro/RetroArch/blob/master/gfx/video_driver.c#L216-L225

This is how it's done for the null driver when returning the drivers data.

Reply:
I see a use case of map keyed by lock ID (pointer to it) and tracking the order in which they are acquired in order to check that some deadlock invariants are respected. While wrapping locks might be a better idea for it, I don't want to check all cores that none of them does something like this and wasting 16-32 bytes per lock (round up and header) is probably nothing to worry about.

Reply:
Never seen lock primitives being used like that and it sounds like a really bad practice.
Identifying locks by their virtual memory address is just asking for trouble, more so when your language (such as C++) has copy and move constructors.

As for memory space, I don't know for which system this is, so I really can't comment, but something this basic, like a lock primitive, could soon add up.

If you really need to map each lock to an unique address, why not:
```
slock_t *slock_new(void)
{
   static uint64_t lock = 0;

   return (slock_t*)lock++;
}
```

I still think it's really bad practice to id lock primitives like that, but in here you don't have the overhead of memory (de)allocation and no heap space wasted.

Reply:
I agree it is a bad practice to do something like I described but cores have wildly varying levels of code quality. I will follow your suggestion but start at 1 for obvious reasons

Reply:
I forgot to change it to 1 before I left, but did it now, and also changed uint64_t to uintptr_t.

Reply:
done